rosidl_runtime_py/rosidl_runtime_py/set_message.py

```diff
@@ -33,4 +38,13 @@ def set_message_fields(msg: Any, values: Dict[str, str]) -> None:
         except TypeError:
             value = field_type()
             set_message_fields(value, field_value)
+        rosidl_type = get_message_slot_types(msg)[field_name]
+        # Check if field is an array of ROS messages
+        if isinstance(rosidl_type, NestedType):
+            if isinstance(rosidl_type.basetype, NamespacedType):
+                field_elem_type = import_message_from_namespaced_type(rosidl_type)
+                for n in range(len(value)):
+                    submsg = field_elem_type()
+                    set_message_fields(submsg, value[n])
```

Comment:
`value[n]` looks weird here? Can you please extend the existing test of `set_message_fields` to cover this case of a nested namespaced type to make sure this works as expected.

Reply:
added test coverage in https://github.com/ros2/rosidl_python/pull/33/commits/ad65970d6bb4bff29a4d36fdc27845fb7d80f3f4